iaso/api/entity.py

```diff
@@ -127,6 +127,64 @@ def get_queryset(self):
             queryset = queryset.filter(name__icontains=search)
         return queryset
 
+    def partial_update(self, request, pk=None):
+        """
+        PATCH /api/entitytypes/{id}/
+        Provides an API to edit an Entity Type
+        Needs iaso_entity_type_write permission
+        """
+
+        if not request.user.has_perm(permission.ENTITY_TYPE_WRITE):
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        name = request.data.get("name", None)
+        if name is None:
+            raise serializers.ValidationError({"name": "This field is required"})
+        try:
+            print(pk)
+            entity_type = EntityType.objects.get(pk=pk)
+            entity_type.name = name
+            entity_type.fields_duplicate_search = request.data.get("fields_duplicate_search", None)
+            entity_type.fields_list_view = request.data.get("fields_list_view", None)
+            entity_type.fields_detail_info_view = request.data.get("fields_detail_info_view", None)
```

Comment:
this should maybe be validated by the serializer.
@SimonDus  @mathvdh Is it OK like this or is it worth refactoring? (Or have I got it wrong?)

Reply:
ya for those two maybe @SimonDus or @mathvdh can give their opinion